stdlib/universe/map.go

```diff
@@ -248,7 +248,10 @@ func (t *mapTransformation) createSchema(b execute.TableBuilder, m values.Object
 	}
 
 	props := make(map[string]semantic.Nature, numProps)
-	for i := 0; i < numProps; i++ {
+	// Deduplicate the properties in the return type.
```

Comment:
Ok so this does work and the error I saw with pandas was a type error unrelated to this. I think this looks good and we will have to add a helper method for this at some point later. This is perfect.